Adding tab management, tab bar and shortcuts

b0yaux opened the pull request:
I just messed out to add tab management, tab bar and shortcuts for tabs.

b0yaux committed:
tabs management and tab bar ui test

b0yaux committed:
fixed UI and messed up with stuff

b0yaux committed:
Update README.md

b0yaux committed:
Update README.md

b0yaux committed:
Update README.md

rlacombe commented:
Awesome thanks! What would be the point of adding multi tabs? I'm aiming for this to feel like a desktop app more than a browser.

b0yaux commented:
Hi !
Personally, i do like to have the ability to have multiple soundcloud pages
opened, probably because it allows to flow through this music network in a
more arborescent way.
I do understand that you value simplicity. In my version, the tab bar is
shown only when you open at least one other tab, inspired by the Terminal
app on macOS (or even Finder).
Do as you wish !
Wish you the best,

On Thu 13 Feb 2025 at 00:51, Romain Lacombe ***@***.***>
wrote:

> Awesome thanks! What would be the point of adding multi tabs? I'm aiming
> for this to feel like a desktop app more than a browser.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rlacombe/SoundCloud-macOS/pull/1#issuecomment-2655105139>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHG7SJNNF3J4N3AYSA6SQNL2PPNAFAVCNFSM6AAAAABW6CAOY6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNJVGEYDKMJTHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

b0yaux committed:
added : flash download with scdl (+debug), enhanced navigation

- Downloader popover using scdl with URL pre-fill. When standing on a track/album/playlist/user page, download super easily using GUI or shortcuts "cmd+D" (show download popover) and "Enter" (start download of pre-filled URL)
- Debug console ladybug for scdl / ffmpeg troubleshooting
- Added preferences panel: for now only light/dark mode settings included there
- "Cmd+T", "Cmd+W", "Cmd+D", "Cmd+L", and "Cmd+," keyboard shortcuts
- Better navigation : Back and Forward navigation buttons with state tracking and proper Open Link in New Tab command in context menu.
- UI/UX polish

b0yaux committed:
reset bundle id as i have no idea what this is used for

b0yaux committed:
Your commit message

b0yaux committed:
update README and screenshot

b0yaux committed:
update README and screenshot

b0yaux committed:
update README and screenshot

b0yaux committed:
update README and screenshot

b0yaux committed:
improved ui, added flac download option, minor bug fixes

b0yaux committed:
improved UI
